platform-api/che-core-api-infrastructure-local/src/main/java/org/eclipse/che/api/local/LocalSshDaoImpl.java

```diff
@@ -0,0 +1,104 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2015 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.local;
+
+import com.google.common.reflect.TypeToken;
+
+import org.eclipse.che.api.core.ConflictException;
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.local.storage.LocalStorage;
+import org.eclipse.che.api.local.storage.LocalStorageFactory;
+import org.eclipse.che.api.ssh.server.model.impl.UsersSshPairImpl;
+import org.eclipse.che.api.ssh.server.spi.SshDao;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+
+/**
+ * In-memory implementation of {@link SshDao}
+ *
+ * @author Sergii Leschenko
+ */
+@Singleton
+public class LocalSshDaoImpl implements SshDao {
+    private final List<UsersSshPairImpl> pairs;
+    private final LocalStorage           sshStorage;
+
+    @Inject
+    public LocalSshDaoImpl(LocalStorageFactory storageFactory) throws IOException {
+        pairs = new ArrayList<>();
+        sshStorage = storageFactory.create("ssh.json");
+    }
+
+    @Override
+    public synchronized void create(UsersSshPairImpl usersSshPair) throws ConflictException {
+        final Optional<UsersSshPairImpl> any = find(usersSshPair.getOwner(), usersSshPair.getService(), usersSshPair.getName());
+        if (any.isPresent()) {
+            throw new ConflictException(format("Ssh pair with name '%s' and service %s already exist.", usersSshPair.getName(),
+                                               usersSshPair.getService()));
+        }
+        pairs.add(usersSshPair);
+    }
+
+    @Override
+    public synchronized UsersSshPairImpl get(String owner, String service, String name) throws NotFoundException {
```

Comment:
BTW synchronization shouldn't be at method level. It's the worst synchronization pattern.
There are new patterns https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/locks/package-summary.html